### PR TITLE
Workaround for cuda 12 issue in cusparse

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

**Review by divyegala — APPROVED** [review]

File: cpp/include/raft/spectral/detail/matrix_wrappers.hpp [hunk 1]
```diff
@@ -241,6 +244,8 @@ struct sparse_matrix_t {
     RAFT_CUSPARSE_TRY(raft::sparse::detail::cusparsespmv(
       cusparse_h, trans, &alpha, matA, vecX, &beta, vecY, spmv_alg, external_buffer.raw(), stream));
 
+    // FIXME: This is a workaround for a cusparse issue being encountered in CUDA 12
```
Comment (benfred) [inline cpp/include/raft/spectral/detail/matrix_wrappers.hpp:247]: Is there a link to a github issue for this problem that we can add here?

**Review by benfred — APPROVED** [review]

**Comment by cjnolet** [comment]
There's not but I'd like to add that in a follow-on. This PR is just applying a patch from Paul Taylor that we had to use for the DLFW containers to workaround a gnarly cuda bug. It's currently holding up cugraph.

**Comment by cjnolet** [comment]
Partially addresses #1509

**Pull request merged**